### PR TITLE
Fix #1589: Naming consistency for domain data provider id

### LINKED ISSUE #1589: Naming consistency for DataProviderIds in domain layer.
In `domain` layer all data provider ids should follow consistent pattern.
We should follow below mentioned style:

 - Name of constant & value should be the same (e.g. `QUESTION_DATA_PROVIDER_ID` and `"question_data_provider_id"`)
 - Constant name should be UPPER_SNAKE_CASE and value should be lower_snake_case
 - Use ID format: `<function_name>_<optional_context>_PROVIDER_ID` where the function_name is the exact function the provider used (expected to be just 1) and optional_context is used to disambiguate either when the ID is used in different functions or if the same function is overloaded

### PR DESCRIPTION
**Pull request by misterjedu**
## Explanation
Fix #1589 Improves naming consistency for all data provider ids in the domain layer.

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

**Comment by misterjedu** [comment]
@rt4914, PTAL

**Comment by rt4914** [comment]
@misterjedu Please update the title to:

`Fix #1589: Naming consistency for domain data provider id`

See point 4 in https://github.com/oppia/oppia-android/wiki#instructions-for-making-a-code-change

File: domain/src/main/java/org/oppia/android/domain/exploration/ExplorationDataController.kt [hunk 1]
```diff
@@ -10,7 +10,8 @@ import org.oppia.android.util.data.DataProviders
 import org.oppia.android.util.system.OppiaClock
 import javax.inject.Inject
 
-private const val EXPLORATION_DATA_PROVIDER_ID = "ExplorationDataProvider"
+private const val EXPLORATION_DATA_CONTROLLER_PROVIDER_ID =
```
Comment (rt4914) [inline domain/src/main/java/org/oppia/android/domain/exploration/ExplorationDataController.kt]: This naming is incorrect.

If you see the issue description, in point 3 it mentions that you have to use following format.

`<function_name>_<optional_context>_PROVIDER_ID`

Which means it should be 

`GET_EXPLORATION_BY_ID_PROVIDER_ID`

File: domain/src/main/java/org/oppia/android/domain/profile/ProfileManagementController.kt [hunk 2]
```diff
@@ -33,29 +33,31 @@ import java.util.Locale
...
-private const val TRANSFORMED_GET_PROFILES_PROVIDER_ID = "transformed_get_profiles_provider_id"
-private const val TRANSFORMED_GET_PROFILE_PROVIDER_ID = "transformed_get_profile_provider_id"
-private const val TRANSFORMED_GET_WAS_PROFILE_EVER_ADDED_PROVIDER_ID =
-  "transformed_was_profile_ever_added_provider_id"
-private const val TRANSFORMED_GET_DEVICE_SETTINGS_PROVIDER_ID =
-  "transformed_device_settings_provider_id"
-private const val ADD_PROFILE_TRANSFORMED_PROVIDER_ID = "add_profile_transformed_id"
-private const val UPDATE_NAME_TRANSFORMED_PROVIDER_ID = "update_name_transformed_id"
+private const val GET_PROFILES_PROVIDER_ID = "get_profiles_provider_id"
+private const val GET_PROFILE_PROVIDER_ID = "xget_profile_provider_id"
```
Comment (rt4914) [inline domain/src/main/java/org/oppia/android/domain/profile/ProfileManagementController.kt]: There is a typo in this:
`private const val GET_PROFILE_PROVIDER_ID = "get_profile_provider_id"`

File: domain/src/main/java/org/oppia/android/domain/profile/ProfileManagementController.kt [hunk 3]
```diff
@@ -33,29 +33,31 @@ import java.util.Locale
...
+private const val GET_WAS_PROFILE_EVER_ADDED_PROVIDER_ID =
```
Comment (rt4914) [inline domain/src/main/java/org/oppia/android/domain/profile/ProfileManagementController.kt:38]: Missing `get_`

`private const val GET_WAS_PROFILE_EVER_ADDED_PROVIDER_ID =
  "get_was_profile_ever_added_provider_id"`

File: domain/src/main/java/org/oppia/android/domain/profile/ProfileManagementController.kt [hunk 4]
```diff
@@ -33,29 +33,31 @@ import java.util.Locale
...
+  "was_profile_ever_added_provider_id"
+private const val GET_DEVICE_SETTINGS_PROVIDER_ID =
```
Comment (rt4914) [inline domain/src/main/java/org/oppia/android/domain/profile/ProfileManagementController.kt:40]: Missing `get_`

`private const val GET_DEVICE_SETTINGS_PROVIDER_ID =
  "get_device_settings_provider_id"`

File: domain/src/main/java/org/oppia/android/domain/profile/ProfileManagementController.kt [hunk 5]
```diff
@@ -33,29 +33,31 @@ import java.util.Locale
...
+  "device_settings_provider_id"
+private const val ADD_PROFILE_PROVIDER_ID = "add_profile_provided_id"
+private const val UPDATE_NAME_PROVIDER_ID = "update_name_provider_id"
 private const val UPDATE_PIN_TRANSFORMED_PROVIDER_ID = "update_pin_transformed_id"
```
Comment (rt4914) [inline domain/src/main/java/org/oppia/android/domain/profile/ProfileManagementController.kt]: Update this to 
`private const val UPDATE_PIN_PROVIDER_ID = "update_pin_id"`

File: domain/src/main/java/org/oppia/android/domain/profile/ProfileManagementController.kt [hunk 6]
```diff
@@ -33,29 +33,31 @@ import java.util.Locale
...
+  "update_profile_avatar_provider_id"
+private const val UPDATE_WIFI_PERMISSION_DEVICE_SETTINGS_PROVIDER_ID =
+  "update_wifi_permission_device_settings_provider_id"
+private const val UPDATE_TOPIC_AUTOMATICALLY_PERMISSION_DEVICE_SETTINGS_PROVIDER_ID =
+  "update_topic_automatically_permission_device_settings_provider_id"
+private const val UPDATE_ALL_DOWNLOAD_ACCESS_PROVIDER_ID =
+  "update_all_download_provider_id"
+private const val LOGIN_TO_PROFILE_PROVIDER_ID = "login_to_profile_provider_id"
+private const val DELETE_PROFILE_PROVIDER_ID = "delete_profile_provider_id"
+private const val SET_CURRENT_PROFILE_ID_PROVIDER_ID = "set_profile_id_provider_id"
```
Comment (rt4914) [inline domain/src/main/java/org/oppia/android/domain/profile/ProfileManagementController.kt]: Nit:
`private const val SET_CURRENT_PROFILE_ID_PROVIDER_ID = "set_current_profile_id_provider_id"`

File: domain/src/main/java/org/oppia/android/domain/question/QuestionAssessmentProgressController.kt [hunk 7]
```diff
@@ -24,8 +24,12 @@ import javax.inject.Inject
 import javax.inject.Singleton
 import kotlin.concurrent.withLock
 
-private const val CURRENT_QUESTION_DATA_PROVIDER_ID = "CurrentQuestionDataProvider"
+private const val CREATE_CURRENT_QUESTION_DATA_PROVIDER_ID =
+  "create_current_question_data_provider_id"
+private const val BEGIN_QUESTION_TRAINING_SESSION_PROVIDER_ID =
+  "begin_question_training_session_provider_id"
 private const val EMPTY_QUESTIONS_LIST_DATA_PROVIDER_ID = "EmptyQuestionsListDataProvider"
```
Comment (rt4914) [inline domain/src/main/java/org/oppia/android/domain/question/QuestionAssessmentProgressController.kt]: Nit fix:
`private const val EMPTY_QUESTIONS_LIST_DATA_PROVIDER_ID = "empty_questions_list_data_provider_id"`

File: domain/src/main/java/org/oppia/android/domain/topic/StoryProgressController.kt [hunk 8]
```diff
@@ -38,15 +38,15 @@ const val RATIOS_EXPLORATION_ID_3 = "tIoSb3HZFN6e"
...
-  "transformed_get_story_progress_list_provider_id"
-private const val TRANSFORMED_GET_TOPIC_PROGRESS_PROVIDER_ID =
-  "transformed_get_topic_progress_provider_id"
-private const val TRANSFORMED_GET_STORY_PROGRESS_PROVIDER_ID =
-  "transformed_get_story_progress_provider_id"
-private const val ADD_STORY_PROGRESS_TRANSFORMED_PROVIDER_ID = "add_story_progress_transformed_id"
-private const val RECENTLY_PLAYED_CHAPTER_TRANSFORMED_PROVIDER_ID =
-  "recently_played_chapter_transformed_id"
+private const val RETRIEVE_TOPIC_PROGRESS_LIST_DATA_PROVIDER_ID =
+  "retrieved_topic_progress_list_data_provider_id"
```
Comment (rt4914) [inline domain/src/main/java/org/oppia/android/domain/topic/StoryProgressController.kt]: Nit fix:
```
private const val RETRIEVE_TOPIC_PROGRESS_LIST_DATA_PROVIDER_ID =
  "retrieve_topic_progress_list_data_provider_id"
```

**Comment by misterjedu** [comment]
@rt4914 , DONE.

File: domain/src/main/java/org/oppia/android/domain/topic/TopicController.kt [hunk 9]
```diff
@@ -64,15 +64,16 @@ private const val FRACTIONS_SUBTOPIC_ID_3 = 3
...
-  "transformed_get_completed_stories_provider_id"
-private const val TRANSFORMED_GET_ONGOING_TOPICS_PROVIDER_ID =
-  "transformed_get_ongoing_topics_provider_id"
-private const val TRANSFORMED_GET_TOPIC_PROVIDER_ID = "transformed_get_topic_provider_id"
-private const val TRANSFORMED_GET_STORY_PROVIDER_ID = "transformed_get_story_provider_id"
-private const val COMBINED_TOPIC_PROVIDER_ID = "combined_topic_provider_id"
-private const val COMBINED_STORY_PROVIDER_ID = "combined_story_provider_id"
+private const val RETRIEVED_QUESTIONS_FOR_SKILLS_ID_PROVIDER_ID =
+  "retrieved_questions_for_skills_id_provider_id"
+private const val GET_COMPLETED_STORy_LIST_PROVIDER_ID =
```
Comment (rt4914) [inline domain/src/main/java/org/oppia/android/domain/topic/TopicController.kt]: Typo fix:
```
private const val GET_COMPLETED_STORY_LIST_PROVIDER_ID =
  "get_completed_story_list_provider_id"
```

File: domain/src/main/java/org/oppia/android/domain/topic/TopicController.kt [hunk 10]
```diff
@@ -64,15 +64,16 @@ private const val FRACTIONS_SUBTOPIC_ID_3 = 3
...
+  "get_completed_story_list_provider_id"
+private const val GET_ONGOING_TOPIC_LIST_PROVIDER_ID =
+  "get_ongoing_topic_list_provider_id"
+private const val GET_TOPIC_PROVIDER_ID = "get_topic_provider_id"
+private const val GET_STORY_PROVIDER_ID = "get_story_provider_id"
+private const val GE_TOPIC_COMBINED_PROVIDER_ID = "get_topic_combined_provider_id"
```
Comment (rt4914) [inline domain/src/main/java/org/oppia/android/domain/topic/TopicController.kt]: Typo fix
```
private const val GET_TOPIC_COMBINED_PROVIDER_ID = "get_topic_combined_provider_id"
```

**Review by rt4914 — CHANGES_REQUESTED** [review]
This is really nice @misterjedu .

I have suggested few changes. 

`Tip 1:` Do not do a force push commit. Check out point 5 here. https://github.com/oppia/oppia-android/wiki#instructions-for-making-a-code-change

`Tip 2`: Make sure you reply to all the comments mentioned by reviewer and the person who has started the comment thread will resolve it. So in this case you don't need to resolve comments.

Thanks.

File: domain/src/main/java/org/oppia/android/domain/profile/ProfileManagementController.kt (hunk 5 above)
Comment (misterjedu) [reply domain/src/main/java/org/oppia/android/domain/profile/ProfileManagementController.kt]: I think it should be 

private const val UPDATE_PIN_PROVIDER_ID = "update_pin_provider_id"

Right?

File: domain/src/main/java/org/oppia/android/domain/exploration/ExplorationDataController.kt (hunk 1 above)
Comment (misterjedu) [reply domain/src/main/java/org/oppia/android/domain/exploration/ExplorationDataController.kt]: Fixed.

File: domain/src/main/java/org/oppia/android/domain/profile/ProfileManagementController.kt (hunk 2 above)
Comment (misterjedu) [reply domain/src/main/java/org/oppia/android/domain/profile/ProfileManagementController.kt]: Fixed.

File: domain/src/main/java/org/oppia/android/domain/profile/ProfileManagementController.kt (hunk 3 above)
Comment (misterjedu) [reply domain/src/main/java/org/oppia/android/domain/profile/ProfileManagementController.kt:38]: Fixed.

File: domain/src/main/java/org/oppia/android/domain/profile/ProfileManagementController.kt (hunk 4 above)
Comment (misterjedu) [reply domain/src/main/java/org/oppia/android/domain/profile/ProfileManagementController.kt:40]: Fixed

File: domain/src/main/java/org/oppia/android/domain/profile/ProfileManagementController.kt (hunk 6 above)
Comment (misterjedu) [reply domain/src/main/java/org/oppia/android/domain/profile/ProfileManagementController.kt]: Fixed

File: domain/src/main/java/org/oppia/android/domain/question/QuestionAssessmentProgressController.kt (hunk 7 above)
Comment (misterjedu) [reply domain/src/main/java/org/oppia/android/domain/question/QuestionAssessmentProgressController.kt]: Fixed

File: domain/src/main/java/org/oppia/android/domain/topic/StoryProgressController.kt (hunk 8 above)
Comment (misterjedu) [reply domain/src/main/java/org/oppia/android/domain/topic/StoryProgressController.kt]: Fixed

File: domain/src/main/java/org/oppia/android/domain/topic/TopicController.kt (hunk 10 above)
Comment (misterjedu) [reply domain/src/main/java/org/oppia/android/domain/topic/TopicController.kt]: Fixed

File: domain/src/main/java/org/oppia/android/domain/topic/TopicController.kt (hunk 9 above)
Comment (misterjedu) [reply domain/src/main/java/org/oppia/android/domain/topic/TopicController.kt]: Fixed

**Comment by misterjedu** [comment]
Thanks, @rt4914 

I have fixed all the corrections you made in my code and will push them now.

**Comment by misterjedu** [comment]
Thanks, @rt4914

I have fixed all the corrections you made in my code and pushed. 

PTAL

File: domain/src/main/java/org/oppia/android/domain/profile/ProfileManagementController.kt (hunk 5 above)
Comment (rt4914) [reply domain/src/main/java/org/oppia/android/domain/profile/ProfileManagementController.kt]: Yes. My mistake.

File: domain/src/main/java/org/oppia/android/domain/question/QuestionAssessmentProgressController.kt (hunk 7 above)
Comment (rt4914) [reply domain/src/main/java/org/oppia/android/domain/question/QuestionAssessmentProgressController.kt]: Thanks for doing this correctly. I appreciate it. @misterjedu

**Review by rt4914 — APPROVED** [review]
Great work. Thanks @misterjedu

**Pull request merged**